documentation/projects/proposals/trust_and_safety/content_report_moderation/20230411-project_proposal_content_report_moderation.md

```diff
@@ -0,0 +1,338 @@
+
+<!-- Detailed descriptions of the features required for the project. Include user stories if you feel they'd be helpful, but focus on describing a specification for how the feature would work with an eye towards edge cases. -->
+
+Requirements are outlined here, with further details below. Each of these
+roughly correspond to a single implementation plan. Requested plans are
+[covered in detail below](#required-implementation-plans).
+
+1. Language is updated to use "sensitive" or "sensitive content" rather than
+   "mature" where possible.
+1. Volunteers outside the core maintainers can access content reports in Django
+   admin and work through a queue of pending reports.
+1. Bulk actions can be taken to deindex or mark sensitive many results.
+1. Use a computer vision API to analyse images on a report to give hints to
+   moderators about the contents of the image so that they can safely triage the
+   report. Images are blurred in Django admin.
+1. Results with confirmed reports are quickly updated to be excluded from
+   non-sensitive API queries.
+1. If volunteer moderators are identified, work with them to establish an
+   initial goal for report response time.
+
+This project does not deal with addressing DMCA reports, though there will
+inevitably be some tooling overlap.
+
+### Update copy to use the more general "sensitive" language (requirement 1)
+
+This will be done in code, Django admin copy, and frontend copy only. The
+related database tables and columns for existing and future reports should
+remain unchanged. While the existing data is small and could be trivially
+transferred to new tables with better names, we can avoid that complication for
+now and defer that as a potential future improvement. We will also update the
+"reason" for reports from `mature` to `sensitive_content`.
+
+### Volunteer moderator Django admin tools and access control (requirement 2)
+
+Django admin already includes granular access control tools. We'll need to scope
+out the creation of a new Django user group with appropriate permissions, the
+management of the users in that group, and how to get volunteer moderators
+through Cloudflare Access. One suggested implementation is to add a new
+`WordPress/openverse-content-moderators` GitHub group able access Django admin.
+A new GitHub group would have the additional benefit of allowing discussions
+about iterations to the moderation processes and tools to include everyone
+actually doing the moderation.
+
+The content report admin views already exist, but need some upgrades to improve
+moderator quality of life:
+
+- Moderator notes, for moderators to supply an explanatory note for their
+  decision. This should be a separate column to the content report description.
+- Improved table organisation by displaying the report ID and making that the
+  clickable column rather than the status (which is confusing).
+- Create a "decision" page that displays all pending and historical reports for
```

Comment:
Would we want this info to be shareable with the content creators?  Do you think we would ever get requests from the authors about why their items were marked as sensitive?

Reply:
I think that falls under the disputes issue that I called out as something we should handle in a separate project. But yes, I think that would be valuable long term, just not part of this baseline project.